compose/config/fields_schema_v2.json

```diff
@@ -0,0 +1,43 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+
+  "type": "object",
+  "properties": {
+    "version": {
+      "enum": [2]
+    },
+    "services": {
+      "type": "object",
+      "patternProperties": {
+        "^[a-zA-Z0-9._-]+$": {
+          "$ref": "fields_schema.json#/definitions/service"
+        }
```

Comment:
I'm hoping we'll be able to refactor the schema so that this service name constraint only exists in once place (instead of both here and the v1 schema).  I think it can wait until later though.